extensions-core/simple-client-sslcontext/src/test/java/org/apache/druid/https/SSLClientConfigTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.https;
+
+import org.apache.druid.metadata.PasswordProvider;
+import org.junit.Test;
+
+import java.lang.reflect.Field;
+
+import static org.junit.Assert.assertEquals;
+
+public class SSLClientConfigTest
+{
+  @Test
+  public void testGetters() throws NoSuchFieldException, IllegalAccessException
+  {
+    SSLClientConfig sslClientConfig = new SSLClientConfig();
+
+    String expectedProtocol = "TLS";
+    String expectedTrustStoreType = "JKS";
+    String expectedTrustStorePath = "/path/to/truststore";
+    String expectedTrustStoreAlgorithm = "SunX509";
+    PasswordProvider expectedTrustStorePasswordProvider = () -> "trustStorePassword";
+    String expectedKeyStorePath = "/path/to/keystore";
+    String expectedKeyStoreType = "JKS";
+    String expectedCertAlias = "alias";
+    PasswordProvider expectedKeyStorePasswordProvider = () -> "keyStorePassword";
+    PasswordProvider expectedKeyManagerPasswordProvider = () -> "keyManagerPassword";
+    String expectedKeyManagerFactoryAlgorithm = "SunX509";
+    Boolean expectedValidateHostnames = true;
+
+    // Use reflection to set the private fields
```

Comment:
Please don't use reflection for this.
Either
- add a constructor
- OR, override the getters to return the expected values
- OR, deserialize the object from a json string

Reply:
Picked up the second option here to override the getters to return the expected values. Pushing changes in the next commit.